### ISSUE TITLE
Image specified for deploy stage for codebuild not used

### ISSUE THREAD

**Issue by eriklz** [awslabs/aws-deployment-framework#194]
It seems image specified for CodeBuild in deploy stage is not used.
This config will use the default Ubuntu image with Python 3.7..1 for both the build and deploy
stages:

```
    default_providers:
      source:
        provider: codecommit
        properties:
          account_id: "123456789012"
      build:
        provider: codebuild
      deploy:
        provider: codebuild
        properties:
          image: AMAZON_LINUX_2
          spec_filename: deployspec.yml
```

and this config will use Amazon Linux 2 in build and deploy stages:

```
    default_providers:
      source:
        provider: codecommit
        properties:
          account_id: "123456789012"
      build:
        provider: codebuild
        properties:
          image: AMAZON_LINUX_2
      deploy:
        provider: codebuild
        properties:
          image: AMAZON_LINUX_2
          spec_filename: deployspec.yml

```

and this config will also use Amazon Linux 2 in build and deploy stages:

```
    default_providers:
      source:
        provider: codecommit
        properties:
          account_id: "123456789012"
      build:
        provider: codebuild
        properties:
          image: AMAZON_LINUX_2
      deploy:
        provider: codebuild
        properties:
          image: STANDARD_2_0
          spec_filename: deployspec.yml

```

In all cases the deployspec.yml file is used for the deploy stage.

**Comment by bundyfx** [comment]
Thanks @eriklz - indeed, this is a bug in the way we are fetching from the input, will include this fix in the next release.

**Comment by bundyfx** [comment]
unfortunately this is still part of 3.0 but will be the first fix of 3.0.1 :)
